python/do_plots.py

```diff
@@ -50,61 +50,142 @@ def formatStatUncHist(hists, name, hstyle=3254):
 
 
 # _____________________________________________________________________________
-def mapHistos(var, label, sel, param, rebin):
-    LOGGER.info('Run plots for var:%s     label:%s     selection:%s',
-                var, label, sel)
-    signal = param.plots[label]['signal']
-    backgrounds = param.plots[label]['backgrounds']
+def load_hists(var: str,
+               label: str,
+               sel: str,
+               config: dict[str, any],
+               rebin: int) -> tuple[dict[str, any], dict[str: any]]:
+    '''
+    Load all histograms needed for the plot
+    '''
+
+    try:
+        signal = config['plots'][label]['signal']
+    except KeyError:
+        signal = {}
+
+    try:
+        backgrounds = config['plots'][label]['backgrounds']
+    except KeyError:
+        backgrounds = {}
 
     hsignal = {}
     for s in signal:
         hsignal[s] = []
-        for f in signal[s]:
-            fin = param.inputDir+f+'_'+sel+'_histo.root'
-            if not os.path.isfile(fin):
-                LOGGER.info('File "%s" not found!\nSkipping it...', fin)
+        for filepathstem in signal[s]:
+            infilepath = config['input_dir'] + filepathstem + '_' + sel + \
+                         '_histo.root'
+            if not os.path.isfile(infilepath):
+                LOGGER.info('File "%s" not found!\nSkipping it...', infilepath)
                 continue
 
-            with ROOT.TFile(fin, 'READ') as tf:
-                h = tf.Get(var)
-                hh = copy.deepcopy(h)
-                hh.SetDirectory(0)
-            scaleSig = 1.
-            try:
-                scaleSig = param.scaleSig
-            except AttributeError:
-                LOGGER.debug('No scale signal, using 1.')
-                param.scaleSig = scaleSig
-            LOGGER.info('ScaleSig: %g', scaleSig)
-            hh.Scale(param.intLumi*scaleSig)
-            hh.Rebin(rebin)
+            with ROOT.TFile(infilepath, 'READ') as infile:
+                hist = copy.deepcopy(infile.Get(var))
+                hist.SetDirectory(0)
+
+                scale = config['scale_sig']
+
+                # Check if histograms were already scaled to lumi
+                try:
+                    scaled = infile.scaled
+                except AttributeError:
+                    LOGGER.error('Input file does not contain scaling '
+                                 'information!\n  %s\nAborting...', infilepath)
+                    sys.exit(3)
```

Comment:
Is this compatible with the histmaker/`build_graph` approach as followed in the tutorial? I.e. `do_plots` now requires the `scaled` parameter to be set in the file, which is written from `run_final_analysis.py`'s `run` function. Is this one executed with `build_graph` as well? 

Also will be a breaking change for anyone having run an analysis with the FW before this PR, and just wanting to redo the plots (would need to reprocess final stage). Perhaps it would be better to assume not-scaled if not set, but give a very clear message what's going on?